util/upload-cookbook.py

```diff
@@ -0,0 +1,241 @@
+import re
+from datetime import datetime
+
+import argparse
+import boto3
+from botocore.exceptions import ClientError
+
+_BACKUP_DIR = "backup"
+_COOKBOOKS_DIR = "cookbooks"
+_bck_date = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+_bck_error_array = set()
+_cp_error_array = set()
+_ls_error_array = set()
```

Comment:
I'd avoid these global sets and I'd move in the main function, since they are used only there.

The _aws_ls function could return the errors (the same is valid for bck and cp functions):
```python
def _aws_ls(s3, region, bucket_name, key):
    errors = set()
    out = s3.list_objects_v2(Bucket=bucket_name, Prefix=key)
    if len(out.get("Contents", [])) > 0:
        errors.add(region)

   return errors
```

Reply:
Sets are re-used inside loops, then a global error message is generated